zipline/pipeline/filters/filter.py

```diff
@@ -533,3 +535,34 @@ class StaticAssets(StaticSids):
     def __new__(cls, assets):
         sids = frozenset(asset.sid for asset in assets)
         return super(StaticAssets, cls).__new__(cls, sids)
+
+
+class NoMissingValues(CustomFilter, SingleInputMixin):
+    """Pipeline filter indicating input term has data for a given window.
+    """
+
+    def __new__(cls,
+                inputs,
+                window_length,
+                mask=NotSpecified):
+
+        if isinstance(inputs[0], Filter):
+            raise ValueError(
+                "Input to filter `NoMissingValues` cannot be a Filter."
+            )
+
+        return super(NoMissingValues, cls).__new__(
+            cls,
+            inputs=inputs,
+            window_length=window_length,
+            mask=mask,
+        )
+
+    def compute(self, today, assets, out, value):
+        if isinstance(value, LabelArray):
```

Comment:
My inner micro-optimizer is slightly twitchy that we're doing this dispatch every day even though we know the dtypes will never change.  In practice though, I think the cost is probably negligible

```
In [28]: %%timeit
    ...: data.any(axis=0)
    ...:
    ...:
100000 loops, best of 3: 3.61 µs per loop

In [30]: %%timeit
    ...: if isinstance(x, int):
    ...:     data.any(axis=0)
    ...: else:
    ...:     data.any(axis=0)
    ...:
100000 loops, best of 3: 3.77 µs per loop
```

0.16 microseconds per factor-dayends up being basically nothing.  For example, for a 10-year pipeline, that adds up to 0.0004 seconds.